### ISSUE TITLE
[a11y]: Carbon Gray100 Sorted columns are white on light color

### ISSUE THREAD

**Issue by dabrad26** [carbon-design-system/carbon#10398]
### Package

carbon-components, carbon-components-react

### Package version

10.49.0

### React version

7.49.1

### Automated testing tool and ruleset

Visual issue

### Description

https://www.carbondesignsystem.com/components/data-table/usage/

In Gray100 active sorted columns are white text on #cacaca background. Making it hard to read and not enough contrast.
![Screen Shot 2022-01-12 at 12 15 33 PM](https://user-images.githubusercontent.com/8028956/149215375-06dc56d6-39da-4aeb-b456-4a97ecd5466a.png)

Not sure how to set Gray100 on CodeSandbox

### CodeSandbox example

https://codesandbox.io/s/nostalgic-williams-c7t6o?file=/index.js

### Steps to reproduce

Click a sort column to make it active. You will notice it turns light background and text is still white. Probably text should be dark or active should be something more like darker color.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by dabrad26** [comment]
I believe the same color issue can be found on selected rows when they are hovered. (Check a row mouse over; it uses same color as selected/hover clickable sort headers)

**Comment by aledavila** [comment]
@dabrad26 Hey I've looked into this and cannot reproduce this. I would suggest try updating to the latest version first v7.52.0 and v10.52.0. This could also be an application level issue with styles. But on our end there is no bug. Thanks. Let me know if you need more help. I'll be closing this for now. 

<img width="995" alt="Screen Shot 2022-01-27 at 3 28 55 PM" src="https://user-images.githubusercontent.com/12533409/151446774-bff5ea42-19f5-44af-acf0-932047ab9225.png">

**Comment by dabrad26** [comment]
You can actually see it on the Carbon website:
https://www.carbondesignsystem.com/components/data-table/usage/

![Screen Shot 2022-01-27 at 1 32 03 PM](https://user-images.githubusercontent.com/8028956/151447115-7b1d73e1-41e7-48b2-afcf-284f56baaa5f.png)
